src/praisonai/praisonai/__init__.py

```
@@ -197,14 +202,33 @@ def __getattr__(name):
+            f"installed but failed to import ({_praisonaiagents_import_error})"
+        ) from _praisonaiagents_import_error
+
     try:
         import praisonaiagents
-        if hasattr(praisonaiagents, name):
-            return getattr(praisonaiagents, name)
-    except ImportError:
-        pass
-    
+    except ModuleNotFoundError:
+        raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)